crates/ty_python_semantic/resources/mdtest/snapshots/instance_layout_conf…_-_Tests_for_ty's_`inst…_-_Built-ins_with_impli…_(f5857d64ce69ca1d).snap

```diff
@@ -12,160 +12,202 @@ mdtest path: crates/ty_python_semantic/resources/mdtest/instance_layout_conflict
    |     `int` instances have a distinct memory layout because of the way `int` is implemented in a C extension
-19 |     str
+21 |     str
    |     --- `str` instances have a distinct memory layout because of the way `str` is implemented in a C extension
-20 | ): ...
+22 | ): ...
    |
 info: rule `instance-layout-conflict` is enabled by default
 
 ```
 
 ```
 error[instance-layout-conflict]: Class will raise `TypeError` at runtime due to incompatible bases
-  --> src/mdtest_snippet.py:22:7
+  --> src/mdtest_snippet.py:24:7
    |
-20 | ): ...
-21 |
-22 | class F(int, str, bytes, bytearray): ...  # error: [instance-layout-conflict]
-   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Bases `int`, `str`, `bytes` and `bytearray` cannot be combined in multiple inheritance
+22 | ): ...
 23 |
-24 | # fmt: on
+24 | class F(int, str, bytes, bytearray): ...  # error: [instance-layout-conflict]
+   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Bases `int`, `str`, `bytes` and `bytearray` cannot be combined in multiple inheritance
+25 |
+26 | @disjoint_base
    |
 info: Two classes cannot coexist in a class's MRO if their instances have incompatible memory layouts
-  --> src/mdtest_snippet.py:22:9
+  --> src/mdtest_snippet.py:24:9
    |
-20 | ): ...
-21 |
-22 | class F(int, str, bytes, bytearray): ...  # error: [instance-layout-conflict]
+22 | ): ...
+23 |
+24 | class F(int, str, bytes, bytearray): ...  # error: [instance-layout-conflict]
    |         ---  ---  -----  --------- `bytearray` instances have a distinct memory layout because of the way `bytearray` is implemented in a C extension
    |         |    |    |
    |         |    |    `bytes` instances have a distinct memory layout because of the way `bytes` is implemented in a C extension
    |         |    `str` instances have a distinct memory layout because of the way `str` is implemented in a C extension
    |         `int` instances have a distinct memory layout because of the way `int` is implemented in a C extension
-23 |
-24 | # fmt: on
+25 |
+26 | @disjoint_base
+   |
+info: rule `instance-layout-conflict` is enabled by default
+
+```
+
+```
+error[instance-layout-conflict]: Class will raise `TypeError` at runtime due to incompatible bases
+  --> src/mdtest_snippet.py:32:7
+   |
+30 |   class H: ...
+31 |
+32 |   class I(  # error: [instance-layout-conflict]
+   |  _______^
+33 | |     G,
+34 | |     H
+35 | | ): ...
+   | |_^ Bases `G` and `H` cannot be combined in multiple inheritance
+36 |
+37 |   # fmt: on
+   |
+info: Two classes cannot coexist in a class's MRO if their instances have incompatible memory layouts
+  --> src/mdtest_snippet.py:33:5
+   |
+32 | class I(  # error: [instance-layout-conflict]
+33 |     G,
+   |     - `G` instances have a distinct memory layout because of the way `G` is implemented in a C extension
+34 |     H
+   |     - `H` instances have a distinct memory layout because of the way `H` is implemented in a C extension
+35 | ): ...
```

Comment:
of course, unlike the classes that we hardcode as being disjoint bases, we don't _know_ that two arbitrary classes decorated as being disjoint bases are marked as such due to the way they're implemented in a C extension. You could add that decorator to a class for whatever reason you like.

It holds true for all of typeshed's disjoint bases, though, and it's _probably_ a good guess in general...?